cmake/SearchForStuff.cmake

```diff
@@ -154,6 +157,11 @@ if(UNIX)
         include_directories(${GTK2_INCLUDE_DIRS})
 	endif()
 
+	if(WAYLAND_FOUND)
```

Comment:
&& WAYLAND_PROTOCOLS_FOUND maybe?

Reply:
`WAYLAND_INCLUDE_DIRS` and `WAYLAND_DEFINITIONS` are actually unrelated to `wayland-protocols`.

The only current usage of `wayland-protocols` is in the implementation of the old (deprecated?) `GSopen` (aka GSopen1) API in which the GS plugin creates its own native window instead of taking over a window given to it by core. It uses the xdg-shell protocol from `wayland-protocols` to handle setting the window title, and as currently unimplemented, resizing. (I focused my development efforts on the non-deprecated `GSopen2` API).

So I think that we should only condition on `wayland-protocols` in the cmake code that uses it directly (only the `WAYLAND_ADD_PROTOCOL_CLIENT` macro's usage(s)).

I'm a cmake noob so I don't know how to get the best error message in case a dev tries to build without having `wayland-protocols`. Would adding an `if(WAYLAND_PROTOCOLS_FOUND)` here be the best way to do it? Or would it generate hard-to-understand error messages since the `.c` and `.h` protocol files would not be generated?
https://github.com/PCSX2/pcsx2/pull/4075/files#diff-273e7a9b69994f653862c746a10ed357c053b6807756d7878de4c06879f2a1adR225

Reply:
your "FindX" stuff should simply error out if a dependency isn't found, and you'll add a check required dep function in the general cmake file.

GSOpen1 will eventually be completely removed so supporting it isn't necessarily necessary, indeed. I don't think we even use it anymore.

Reply:
We do use GSOpen1 for GSDumpGUI